src/modules/graphics/Polyline.cpp

```diff
@@ -185,6 +192,7 @@ void MiterJoinPolyline::renderEdge(std::vector<Vector2> &anchors, std::vector<Ve
 	if (fabs(det) / (len_s * len_t) < LINES_PARALLEL_EPS && Vector2::dot(s, t) > 0)
 	{
 		// lines parallel, compute as u1 = q + ns * w/2, u2 = q - ns * w/2
+		assert(ns == ns); //NaN check
```

Comment:
We don't generally use C's `assert` in love's code because it's not too helpful for users since it won't get converted to a Lua error (whereas throwing a `love::Exception` will).

Should this throw an exception in that case? What user input will cause NaNs here, given the degenerate segment check above?

Reply:
The asserts are more for proving that the fix works, in that you can remove the changed lines, and observe that the assert was hit.

For now there would be no other way for the NaNs so sneak in, but if someone changed/reorginized/refactored the surrounding code in the future, the exceptions are a reminder that this issue is possible.

As well, if there was a testSuite, then the exceptions are a nice direct way of testing for the functionality, its just 'Run the code' and the asserts will terminate, instead of someway testing that the line segments were drawn with a specific set of points.

-------------------------------------

I can change them to love:Exception if you'd like, or remove them if you prefer